core/services/pipeline/task.eth_tx.go

```diff
@@ -113,11 +126,29 @@ func (t *ETHTxTask) Run(_ context.Context, vars Vars, inputs []Result) (result R
 	// NOTE: This can be easily adjusted later to allow job specs to specify the details of which strategy they would like
 	strategy := bulletprooftxmanager.SendEveryStrategy{}
 
-	_, err = t.txManager.CreateEthTransaction(t.db, fromAddr, common.Address(toAddr), []byte(data), uint64(gasLimit), &txMeta, strategy)
+	var confirmations null.Uint32
+	if minConfirmations > 0 {
+		// Store the task run ID so we can resume the pipeline when tx is confirmed
+		confirmations = null.Uint32From(uint32(minConfirmations))
+		txMeta.PipelineTaskRunID = &t.id
+		txMeta.MinConfirmations = confirmations
+	}
+
+	_, err = t.txManager.CreateEthTransaction(t.db, bulletprooftxmanager.NewTx{
+		FromAddress:    fromAddr,
+		ToAddress:      common.Address(toAddr),
+		EncodedPayload: []byte(data),
+		GasLimit:       uint64(gasLimit),
+		Meta:           &txMeta,
+		Strategy:       strategy,
+	})
 	if err != nil {
 		return Result{Error: errors.Wrapf(ErrTaskRunFailed, "while creating transaction: %v", err)}
 	}
-	// TODO(spook): once @archseer's "async jobs" work is merged, return the tx hash of
-	// the successful EthTxAttempt
+
+	if minConfirmations > 0 {
+		return Result{Error: ErrPending}
```

Comment:
Note we may need some follow up UI work to make this a yellow state, as is I think itd show up as red and people would assume something bad happened

Reply:
This is just a sentinel value and it gets cleared out in the scheduler:

https://github.com/smartcontractkit/chainlink/blob/a940234f2db729c9c45153e31f8b2c16a3cd37f3/core/services/pipeline/scheduler.go#L168-L172

It won't get persisted as an error

Reply:
I'd like to change this eventually by having `task.Run()` calls return a meta struct with some extra information

Reply:
Can we leave a comment to this effect along with a TODO pointing to a clubhouse followup story